Firestore/core/test/unit/util/byte_stream_test.cc

```diff
@@ -21,19 +21,93 @@ namespace firestore {
 namespace util {
 namespace {
 
+std::string LargeString() {
+  std::string source;
+  for (int i = 0; i < 10000; ++i) {
+    source +=
+        "{키스의 고유조건은 입술끼리 만나야 하고 특별한 기술은 필요치 "
+        "않다}"
+
+        "{သီဟိုဠ်မှ ဉာဏ်ကြီးရှင်သည် အာယုဝဍ္ဎနဆေးညွှန်းစာကို "
+        "ဇလွန်ဈေးဘေးဗာဒံပင်ထက် အဓိဋ္ဌာန်လျက် "
+        "ဂဃနဏဖတ်ခဲ့သည်။เป็นมนุษย์สุดประเสริฐเลิศคุณค่า}"
+
+        "{กว่าบรรดาฝูงสัตว์เดรัจฉาน จงฝ่าฟันพัฒนาวิชาการ อย่าล้า"
+        "งผลาญฤๅเข่นฆ่าบีฑาใคร ไม่ถือโทษโกรธแช่งซัดฮึดฮัดด่า "
+        "หัดอภัยเหมือนกีฬาอัชฌาสัย ปฏิบัติประพฤติกฎกำหนดใจ "
+        "พูดจาให้จ๊ะ ๆ จ๋า ๆ น่าฟังเอยฯ}";
+  }
+  return source;
+}
+
+uint64_t kMaxIteration = 10000 * 10000;
+
 TEST_P(ByteStreamTest, ReadsStringStream) {
   auto stream = stream_factory_->CreateByteStream("ok");
 
   auto result = stream->ReadUntil('o', 10);
-  EXPECT_EQ(result.result().ValueOrDie(), "");
+  EXPECT_EQ(result.ValueOrDie(), "");
   EXPECT_FALSE(result.eof());
 
   result = stream->ReadUntil('k', 10);
-  EXPECT_EQ(result.result().ValueOrDie(), "o");
+  EXPECT_EQ(result.ValueOrDie(), "o");
   EXPECT_FALSE(result.eof());
 
   result = stream->Read(10);
-  EXPECT_EQ(result.result().ValueOrDie(), "k");
+  EXPECT_EQ(result.ValueOrDie(), "k");
+  EXPECT_TRUE(result.eof());
+}
+
+TEST_P(ByteStreamTest, ReadsEmptyStringStream) {
+  auto stream = stream_factory_->CreateByteStream("");
+
+  auto result = stream->ReadUntil('o', 10);
+  EXPECT_EQ(result.ValueOrDie(), "");
+  EXPECT_TRUE(result.eof());
+
+  result = stream->Read(10);
+  EXPECT_EQ(result.ValueOrDie(), "");
+  EXPECT_TRUE(result.eof());
+}
+
+TEST_P(ByteStreamTest, ReadUtilReadsDelimString) {
+  auto stream = stream_factory_->CreateByteStream("{{{{");
+
+  auto result = stream->ReadUntil('{', 10);
```

Comment:
Can you call `ReadUntil('{', 10);` once more just to make sure it still doesn't read anything by mistake?

Reply:
Done.